src/main/java/de/tum/in/www1/artemis/service/ParticipationService.java

```diff
@@ -171,9 +172,15 @@ else if (exercise instanceof QuizExercise || exercise instanceof ModelingExercis
                 // in case the participation was finished before, we set it to initialized again so that the user sees the correct button "Open modeling editor" on the client side
                 participation.setInitializationState(INITIALIZED);
             }
+
             if (!Optional.ofNullable(participation.getInitializationDate()).isPresent()) {
                 participation.setInitializationDate(ZonedDateTime.now());
             }
+
+            if (participation.getId() == null || !submissionRepository.existsByParticipationId(participation.getId())) {
+                // initialize a modeling or text submission (depending on the exercise type), it will not do anything in the case of a quiz exercise
```

Comment:
What happens with programming exercises and file upload exercises?

Reply:
We will only get into this `else if` block for quiz, modeling and text exercises:
`else if (exercise instanceof QuizExercise || exercise instanceof ModelingExercise || exercise instanceof TextExercise)`

Reply:
As discussed, I changed it to handle file upload exercises as well.